src/saving/SaveInformation.cs

```diff
@@ -35,7 +35,10 @@ public enum SaveType
 
     public string Platform { get; set; } = FeatureInformation.GetOS();
 
-    public string Creator { get; set; } = Environment.UserName;
+    public string Creator { get; set; } =
+        Settings.Instance.CustomUsernameEnabled ?
```

Comment:
I think this should also check that CustomUsername is not null, if it is then Environment.UserName should be used.